polaris-react/src/components/Navigation/components/Item/Item.tsx

```diff
@@ -101,6 +45,11 @@ export function Item({
   expanded,
   shouldResizeIcon,
   truncateText,
+  addLine,
+  addHoverLine,
+  addHoverPointer,
```

Comment:
Since Item is called recursively I had to add these new props to inform the component what classnames it should apply for a given Item. This was the least amount of props added I could come up with. Let me know if you have different ideas that might work too.